src/MetaMorpho.sol

```diff
@@ -638,6 +667,29 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
         return MORPHO.expectedSupplyBalance(marketParams, address(this));
     }
 
+    /// @dev Returns the vault's assets balance the market defined by `marketParams`.
+    /// @dev To use when the function must not revert.
+    /// @return The vault's balance on the market defined by `marketParams`.
+    /// @return Whether the call succeeded or not. This information is needed when computing the suppliable amount.
+    function _simulateSupplyBalance(MarketParams memory marketParams) internal view returns (uint256, bool) {
+        try this.getExpectedSupplyBalance(marketParams) returns (uint256 balance) {
+            return (balance, true);
+        } catch {
+            return (0, false);
+        }
+    }
+
+    /// @dev Returns the vault's share balance the market defined by `id`.
+    /// @dev To use when the function must not revert.
+    /// @return Whether the call succeeded or not.
+    function _simulateSupplyShares(Id id) internal view returns (uint256, bool) {
+        try this.getSupplyShares(id) returns (uint256 shares) {
```

Comment:
perhaps we'd like to do a delegate call instead of a call to preserve the correct context